File: src/main/java/com/support_App/model/UserU.java
package com.support_App.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.CascadeType;
import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;

import java.util.List;

@Entity
@DiscriminatorValue("UserU")

public class UserU extends User{
    @OneToMany(mappedBy = "userU", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    @JsonIgnore
    private List<SupportTicket> supportTickets;

    @OneToMany(mappedBy = "userU", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    @JsonIgnore
    private List<Breakdown> breakdowns;


}

File: IT_FrontEnd/src/app/model/UserU/userU.ts
import {User} from "../User/user";
import {Role} from "../../enums/role";
import {SupportTicket} from "../SupportTicket/support-ticket";

export class UserU extends User{

  supportTickets: SupportTicket[] = [];

  constructor() {
    super();
    this.role = Role.UserU

  }

}
